### ISSUE TITLE
"Deadlock found when trying to get lock" exception when hitting authorize endpoint.

### ISSUE THREAD

**Issue by pz12** [mitreid-connect/OpenID-Connect-Java-Spring-Server#1492]
In the HSPC implementation of the mitreid connect server, we are running into an issue when redirecting to the /authorize endpoint. Sometimes (not often enough to faithfully reproduce), the error shown in the image below appears. Has anyone else run into this or is this a problem with our configuration?

Thanks

<img width="1644" alt="Screen Shot 2019-06-25 at 8 39 33 AM" src="https://user-images.githubusercontent.com/21222506/60107810-c61e4480-9724-11e9-8aba-d9b75cbe8d96.png">

**Comment by lannamalai** [comment]
Yes, we too face the issue while making /authorize call. Especially on inserting AUTHENTICATION_HOLDER related entities, did we have any solution for this issue

`Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.5.1.v20130918-f2b9fc5): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: java.sql.BatchUpdateException: (conn:14) Deadlock found when trying to get lock; try restarting transaction
Error Code: 1213
Query: DataModifyQuery(sql="INSERT INTO AUTHENTICATION_HOLDER_EXTENSION (owner_id, val, extension) VALUES (?, ?, ?)")
`

**Comment by nschwertner** [comment]
Logica has been experiencing the same problem with MitreID

![1e04f475-6d9a-4bde-87ee-84e015d88b7c](https://user-images.githubusercontent.com/752839/110043150-32997580-7d47-11eb-906e-9ef5d04f68b7.png)
